### PR TITLE
Suggested additions to Mabel's PR

### PR DESCRIPTION
**Pull request by bsb808**
* Moving VIM to the end of the Dockerfile.  The stuff at the beginning is more required - trying to keep optional stuff towards the end.
* Updates the build and join scripts for docker management to be consistent with VRX and other projects. 
* Updates the VCS repo list to assist in environment setup.
* Removing https://github.com/Field-Robotics-Lab/glider_hybrid_whoi from VCS list.  There seems to be a problem with - see https://github.com/Field-Robotics-Lab/glider_hybrid_whoi/issues/57

### CODE REVIEW

File: extras/repos/dave_sim.repos [hunk 1]
```diff
@@ -1,65 +1,37 @@
 repositories:
-  dave/dave:
-    type: git
-    url: https://github.com/Field-Robotics-Lab/dave.git
-    version: master
-
-  dave/acoustic_msgs:
+  acoustic_msgs:
```
Comment (mabelzhang) [inline extras/repos/dave_sim.repos:2]: Rename to `hydrographic_msgs`? The URL is being redirected there

File: extras/docker/dave_noetic/Dockerfile [hunk 2]
```diff
@@ -77,6 +76,12 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
     ros-${DIST}-rqt \
     ros-${DIST}-rqt-common-plugins \
     ros-${DIST}-geographic-msgs \
+
```
Comment (mabelzhang) [inline extras/docker/dave_noetic/Dockerfile:79]: Is this extra line intentional?

**Review by mabelzhang — COMMENTED** [review]
Does this PR replace #114?

I didn't realize there's a `.repos` file. Is this somewhere in the wiki? I'm thinking it should go somewhere on the Installation page, along with my comments in #119.

File: extras/repos/dave_sim.repos [hunk 3]
```diff
@@ -1,65 +1,37 @@
...
-    version: master
-
-  uuv_sim/rexrov2:
+  rexrov2:
     type: git
     url: https://github.com/uuvsimulator/rexrov2.git
     version: master
-
-  uuv_sim/uuv_manipulators:
+  uuv_manipulators:
```
Comment (mabelzhang) [inline extras/repos/dave_sim.repos:30]: I believe this URL also needs to be `Field-Robotics-Lab`, and `version: noetic_updates`.

File: extras/repos/dave_sim.repos [hunk 4]
```diff
@@ -1,65 +1,37 @@
...
     url: https://github.com/uuvsimulator/uuv_manipulators.git
     version: master
-
-  uuv_sim/uuv_simulator:
-    type: git
-    url: https://github.com/uuvsimulator/uuv_simulator.git
-    version: master
-
-  whoidsl/ds_msgs:
+  uuv_simulator:
```
Comment (mabelzhang) [inline extras/repos/dave_sim.repos:34]: I think this needs `version: noetic_updates`

File: extras/repos/dave_sim.repos [hunk 5]
```diff
@@ -1,65 +1,37 @@
...
-    url: https://github.com/Field-Robotics-Lab/frl_msgs.git
-    version: master
-
-  dave/gtri_based_sonar:
+    url: git@github.com:Field-Robotics-Lab/dave.git
+    version: bsb/update_2_fix_dockerfile
+  ds_msgs:
     type: git
-    url: https://github.com/Field-Robotics-Lab/gtri_based_sonar.git
+    url: https://bitbucket.org/whoidsl/ds_msgs.git
```
Comment (mabelzhang) [inline extras/repos/dave_sim.repos:12]: This `url` might need to be `Field-Robotics-Lab`
I had an error about `ds_sensor_msgs` when I compiled using the BitBucket repo.

File: extras/repos/dave_sim.repos [hunk 6]
```diff
@@ -1,65 +1,37 @@
...
     version: master
-
-  dave/nps_uw_sensors_gazebo:
+  ds_sim:
     type: git
-    url: https://github.com/Field-Robotics-Lab/nps_uw_sensors_gazebo.git
+    url: https://bitbucket.org/whoidsl/ds_sim.git
```
Comment (mabelzhang) [inline extras/repos/dave_sim.repos:16]: Same for this `url`, `Field-Robotics-Lab`

**Review by mabelzhang — COMMENTED** [review]
I was only able to compile when I switched to DAVE's fork of `ds_sim`, `ds_msgs`, `uuv_simulator` (`noetic_updates` branch) and `uvv_manipulators` (`noetic_updates` branch). There are some very recent (few days) changes to these forks.
Were you able to build using the upstream repos?
@dtdavi1 are the latest updates to `ds_sim` and `ds_msgs` needed for the Noetic build?

File: extras/repos/dave_sim.repos (hunk 3 above)
Comment (dtdavi1) [reply extras/repos/dave_sim.repos:30]: None of my updates to the ds_sim and ds_msgs repos should affect Noetic one way or the other.  I still can't test with Noetic, though (still can't get the Docker image to work), so I can't guarantee anything.  @crvogt did most of the Noetic updates, so any commits that he made to those repos might be Noetic related.

**Comment by bsb808** [comment]
> Does this PR replace #114?
> 
> I didn't realize there's a `.repos` file. Is this somewhere in the wiki? I'm thinking it should go somewhere on the Installation page, along with my comments in #119.

The use of vcs and the repos file was in the wiki, but the location was not obvious.  I moved the content here: https://github.com/Field-Robotics-Lab/dave/wiki/Clone-Dave-Repositories

File: extras/repos/dave_sim.repos (hunk 1 above)
Comment (bsb808) [reply extras/repos/dave_sim.repos:2]: Yes - corrected in e8d1b987eff6e2978914f4db5789d8f5f6bd9b73

File: extras/docker/dave_noetic/Dockerfile (hunk 2 above)
Comment (bsb808) [reply extras/docker/dave_noetic/Dockerfile:79]: Corrected in 5379d335e9783d1f97576045c97398be5fd9009f

**Comment by bsb808** [comment]
> Does this PR replace #114?

Apologies - I lost track of #114.  I think there are still outstanding issues in #114.  Let's address this PR, then we can merge your changes in #118.  Finally I'll rebase #114 and make any adjustments.

File: extras/repos/dave_sim.repos (hunk 3 above)
Comment (bsb808) [reply extras/repos/dave_sim.repos:30]: Very good point.  I think we've got out of sync w.r.t. which repos and which forks we are officially supporting.  I think @crvogt and @M1chaelM should be able to tell us what is the most recent for the change to Noetic.  I'll get that info and make any updates to this PR.

**Comment by bsb808** [comment]
This is currently on hold - I expect that @M1chaelM 's work on a general docker image - and my work on VCS support - will alleviate the need for this PR.

**Comment by woensug-choi** [comment]
Compile error resolved and merged at `glider_hybrid_whoi` repo. It should be ok to include.

**Comment by woensug-choi** [comment]
The `glider_hybrid_whoi` repo include massive static bathymetry tiles of the Buzzards Bay. Since we don't want static data to be included in the mainstream `dave` repo (https://github.com/Field-Robotics-Lab/dave/issues/107), **excluding the glider repo seems reasonable**. I will make and host a dataset somewhere and make tutorials accordingly.

**Comment by M1chaelM** [comment]
Docker related changes in this PR have been addressed elsewhere and are no longer needed.

**Comment by bsb808** [comment]
Agreed - closing this b/c we have address the VCS and docker issues elsewhere.

**Pull request closed without merging**